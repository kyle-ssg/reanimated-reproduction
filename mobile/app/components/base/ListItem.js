import React, { Component } from 'react';
import propTypes from 'prop-types';
import ReactNative, { Platform, TouchableNativeFeedback } from 'react-native';

const ListItem = class extends Component {
  static displayName = 'ListItem';

  static propTypes = {
    accessible: propTypes.bool,
    accessibilityLabel: propTypes.string,
    icon: propTypes.node,
    children: propTypes.oneOfType([
      propTypes.arrayOf(propTypes.node),
      propTypes.node,
    ]).isRequired,
    delay: propTypes.number,
    index: propTypes.number,
    onPress: propTypes.func,
    style: propTypes.any,
    disabled: propTypes.bool,
  };

  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const content = (
        <View style={[Styles.ph5, this.props.style]}>
            <Row>
                {this.props.icon}
                <View
                  style={[
                    this.props.disabled && Styles.listItemDisabled,
                    Styles.liContent,
                    { backgroundColor: 'transparent' },
                  ]}
                >
                    {this.props.children}
                </View>
            </Row>
        </View>
    );

    return this.props.onPress ? (
        <View style={this.props.style || Styles.listItem}>
            {Platform.OS === 'android' ? (
                <TouchableNativeFeedback
                  accessible={this.props.accessible}
                  accessibilityLabel={this.props.accessibilityLabel}
                  onPress={this.props.disabled ? null : this.props.onPress}
                  background={TouchableNativeFeedback.SelectableBackgroundBorderless()}
                >
                    {content}
                </TouchableNativeFeedback>
            ) : (
                <TouchableOpacity
                  accessible={this.props.accessible}
                  accessibilityLabel={this.props.accessibilityLabel}
                  activeOpacity={0.8}
                  onPress={this.props.disabled ? null : this.props.onPress}
                >
                    {content}
                </TouchableOpacity>
            )}
        </View>
    ) : (
        <View
          style={[this.props.style || Styles.listItem]}
        >
            {content}
        </View>
    );
  }
};
ListItem.defaultProps = {
  onPress: null,
  delay: 0,
};

module.exports = ListItem;
