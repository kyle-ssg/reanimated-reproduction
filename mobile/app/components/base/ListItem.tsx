import React from 'react'
import { Platform, TouchableNativeFeedback } from 'react-native'

interface Props {
  accessible?: boolean
  accessibilityLabel?: string
  active?: boolean
  icon?: React.ReactNode
  rightElement?:
    | React.ComponentType<any>
    | React.ReactElement<any, string | React.JSXElementConstructor<any>>
  testID?: string
  children: [React.ReactNode] | React.ReactNode
  delay?: number
  index?: number
  onPress?: () => void
  style?: any
  disabled?: boolean
  isEditable?: boolean
}

export const ListItem: React.FC<Props> = ({
  accessible,
  active,
  testID,
  accessibilityLabel,
  style,
  icon,
  rightElement,
  disabled,
  children,
  onPress = null,
  delay = 0,
}) => {
  const content = (
    <View style={[disabled && Styles.listItemDisabled]}>
      <Container style={[Styles.listItemBorder, Styles.pb10]}>
        <Row>
          {icon ? <View style={[Styles.pr10]}>{icon}</View> : null}
          <View>{children}</View>
          <Flex style={[Styles.alignItemsEnd]}>{rightElement}</Flex>
        </Row>
      </Container>
    </View>
  )

  return onPress ? (
    <View style={[Styles.listItem, style, active && Styles.listItemActive]}>
      {Platform.OS === 'android' ? (
        <TouchableNativeFeedback
          testID={testID}
          accessible={accessible}
          accessibilityLabel={accessibilityLabel}
          onPress={disabled ? null : onPress}
          background={TouchableNativeFeedback.SelectableBackgroundBorderless()}
        >
          {content}
        </TouchableNativeFeedback>
      ) : (
        <TouchableOpacity
          testID={testID}
          accessible={accessible}
          accessibilityLabel={accessibilityLabel}
          activeOpacity={0.8}
          onPress={disabled ? null : onPress}
        >
          {content}
        </TouchableOpacity>
      )}
    </View>
  ) : (
    <View style={[Styles.listItem, style, active && Styles.listItemActive]}>
      {content}
    </View>
  )
}

export default ListItem
