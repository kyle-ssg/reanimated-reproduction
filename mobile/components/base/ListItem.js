var Listitem = Component({
	displayName: 'ListItem',
	getDefaultProps: function () {
		return {
			onPress: null,
			text: null,
			delay: 0,
			animationProps: {
				animation: 'basicListEntrance',
				duration: 250,
				easing: 'ease-out-quart',
			},
			underlayColor: colour.inputBackground,
		};
	}, render: function () {
		content = (
			<Row>
				{this.props.icon}
				<View
					style={[this.props.disabled && Styles.listItemDisabled, Styles.liContent, {backgroundColor: 'transparent'}]}>
					{this.props.children}
				</View>
			</Row>
		);
		const animationProps = Object.assign(this.props.animationProps, {
			delay: this.props.delay + 10 + ((Number(this.props.index || 0)) * 20),
		})
		return (
			this.props.onPress ?
				<Animatable.View
					style={this.props.style || Styles.listItem}
					{...this.props.animationProps}
				>
					<TouchableOpacity
						activeOpacity={0.8}
						onPress={this.props.disabled ? null : this.props.onPress}>
						{content}
					</TouchableOpacity>
				</Animatable.View>
				: <Animatable.View
					{...this.props.animationProps}
					style={[this.props.style || Styles.listItem]}>{content}</Animatable.View>
		);
	}
});
module.exports = Listitem;
