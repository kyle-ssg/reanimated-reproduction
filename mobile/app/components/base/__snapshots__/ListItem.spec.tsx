import * as React from 'react'
import ListItem from '../ListItem'
import renderer from 'react-test-renderer'

describe('<ListItem />', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<ListItem />)
    expect(tree).toMatchSnapshot()
  })
})
