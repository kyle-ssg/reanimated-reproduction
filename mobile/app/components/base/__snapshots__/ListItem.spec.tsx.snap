// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "alignSelf": "stretch",
        "backgroundColor": "white",
        "borderBottomWidth": 0.5,
        "minHeight": 44,
        "paddingBottom": 15,
        "paddingTop": 15,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "paddingRight": 8,
          }
        }
      />
      <View
        style={
          Array [
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
